### PR TITLE
Add CLI option to `sea-orm-cli` to skip primary keys with serde on generation of entities

### PR DESCRIPTION
**Pull request by Witcher01**
## PR Info

- Closes https://github.com/SeaQL/sea-orm/issues/841

## Adds

- [ ] Command line argument `--skip-primary-key-deserialization` to `sea-orm-cli` for the option to skip deserialization of primary keys with serde by adding `#[serde(skip_deserialization)]` to the primary key

As the comment says, this adds the option to skip serializing primary keys with serde in entities generated by `sea-orm-cli`.
The command line flag name is a bit long, so feel free to suggest a different name for it.
The implementation could use some work to feel better, so I'll take suggestions for this as well.

As of now `clap` doesn't treat the new argument as only being allowed to be set when `--with-serde` with a value other than `none` is used.

### CODE REVIEW

**Comment by Witcher01** [comment]
Right, I didn't look at the tests. This might take a bit of time

File: sea-orm-codegen/src/entity/writer.rs [hunk 1]
```diff
@@ -594,8 +615,15 @@ impl EntityWriter {
...
-                        #[sea_orm(#ts)]
+                    if skip_pk_deserialization {
+                        quote! {
+                            #[sea_orm(#ts)]
+                            #[serde(skip_deserialization)]
+                        }
+                    } else {
+                        quote! {
+                            #[sea_orm(#ts)]
+                        }
```
Comment (billy1624) [inline sea-orm-codegen/src/entity/writer.rs:626]: Hey @Witcher01, sorry for the delay! Thanks for the great contributions!! Looks good overall :)

I have a minor nitpick. Please check if the refactoring make any sense
- https://github.com/Witcher01/sea-orm/compare/skip-serde-deserialization...billy1624:pr/846

Comment (Witcher01) [reply sea-orm-codegen/src/entity/writer.rs:626]: No worries! I'm not sure what you mean, my commit looks good to me, tests run, and running it gives me the right results, too.
Could you elaborate? Thanks.

**Comment by Witcher01** [comment]
I just noticed that skipping the primary key deserialization isn't added in the expanded formats. I'll fix that and push a new commit when I'm done. ETA this evening, hopefully

**Comment by billy1624** [comment]
Thanks! @Witcher01

File: sea-orm-codegen/src/entity/writer.rs (hunk 1 above)
Comment (billy1624) [reply sea-orm-codegen/src/entity/writer.rs:626]: I don't want to keep a mutable `skip_pk_deserialization` and that's why refactored the code :P

**Pull request closed without merging**